app/src/state/editor/edit.ts

```diff
@@ -191,15 +193,73 @@ export const paste = createAsyncActionWithReducer<EditorState, void, Document>(
     fulfilled: (state, payload) => {
       state.selection = null;
       state.document.sources = { ...state.document.sources, ...payload.sources };
-      const beforeSlice = DocumentGenerator.fromParagraphs(state.document.content).filter(
-        (item) => item.absoluteStart + item.length <= state.currentTimePlayer
+
```

Comment:
this function is kinda long and messy but it was hard for me to come up with an easier way of doing this. I have a few Ideas on improving this but it requires a new file format. So for now I would go with this one and not invest much time?

Reply:
👍